src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V1/DownloadedPackagesSQLLiteStoreV1.swift

```diff
@@ -155,16 +174,36 @@ extension DownloadedPackagesSQLLiteStoreV1: DownloadedPackagesStoreV1 {
 
 			guard deleteHours() else {
 				self.database.rollback()
+				self.completeAsyncWithError(completion: completion, errorCode: database.lastErrorCode())
 				return
 			}
 			guard insertDay() else {
 				self.database.rollback()
+				self.completeAsyncWithError(completion: completion, errorCode: database.lastErrorCode())
 				return
 			}
 
 			self._commit()
+
+			self.completeAsync(completion: completion)
+		}
+
+	}
+
+	private func completeAsyncWithError(completion: ((SQLiteError?) -> Void)?, errorCode: Int32) {
+		queue.async {
+			if let error = SQLiteError(rawValue: errorCode) {
+				completion?(error)
+			} else {
+				completion?(.unknown)
+			}
 		}
+	}
 
+	private func completeAsync(completion: ((SQLiteError?) -> Void)?) {
+		DispatchQueue.global().async {
```

Comment:
`global` is not the main thread! Did you mean `main`:

```suggestion
		DispatchQueue.main.async {
```

Reply:
Refactored it as discussed